assets/src/components/form/inputs/checkbox-input.js

```diff
@@ -12,110 +12,93 @@ import { OptionInputs } from './base/option-inputs';
 import { OptionLabelSize } from './base/option-label-size';
 
 /**
- * generates one or more html checkbox inputs
+ * Generates one or more html checkbox inputs
  *
- * @constructor
- * @param {string} htmlId
- * @param {string} htmlClass
- * @param {string} value
- * @param {Array} options
- * @param {Function} onChange
- * @param {boolean} btnGroup
- * @param {string} helpTextID
- * @param {Object} dataSet
- * @param {string} inputWidth
- * @param {Object} attributes
- * @return {string} rendered date name form row
+ * @function
+ * @param {Object} props
+ * @return {string} rendered checkbox inputs
  */
-export class CheckboxInput extends Component {
-	static propTypes = {
-		htmlId: PropTypes.string.isRequired,
-		htmlClass: PropTypes.string,
-		value: PropTypes.oneOfType( [
-			PropTypes.array,
-			PropTypes.number,
-			PropTypes.string,
-		] ),
-		options: PropTypes.array,
-		onChange: PropTypes.func,
-		btnGroup: PropTypes.bool,
-		inputWidth: PropTypes.string,
-		helpTextID: PropTypes.string,
-		dataSet: PropTypes.object,
-	};
+export const CheckboxInput = ( {
+	name,
+	htmlId,
+	options,
+	required,
+	helpTextID,
+	btnGroup = true,
+	inputWidth = '',
+	...attributes
+} ) => {
+	const [ checked, setChecked ] = useState( resetState() );
+	let { htmlClass } = attributes;
 
-	constructor( props ) {
-		super( props );
-		this.state = {
-			checked: this.resetState(),
-		};
-	}
+	htmlClass = required ?
+		`${ htmlClass } ee-checkbox-group-required` :
+		htmlClass;
+	const labelClass = btnGroup ?
+		'btn btn-primary' :
+		'ee-checkbox-label-after' + OptionLabelSize( options );
+	let divClass = btnGroup ? 'btn-group' : 'ee-checkbox-group';
+	divClass = inputWidth ?
+		`${ divClass } ee-input-width-${ inputWidth }` :
+		divClass;
 
 	/**
 	 * @function
 	 * @return {Object} state
 	 */
-	resetState = () => {
+	const resetState = () => {
 		return OptionCheckedState(
-			this.props.htmlId,
-			this.props.options,
-			this.props.initialValue,
+			htmlId,
+			options,
+			attributes.initialValue,
```

Comment:
should this be introduced as a prop var as well?
or do we still need it passed through as part of the attributes prop?

Reply:
I will need to check whether it's required to be passed

Reply:
Yes, you are right. `initialValue` is not needed.